x/projects/keeper/project.go

```diff
@@ -49,6 +48,35 @@ func (k Keeper) GetProjectForDeveloper(ctx sdk.Context, developerKey string, blo
 	return project, nil
 }
 
+func (k Keeper) AddKeysToProject(ctx sdk.Context, projectID string, adminKey string, projectKeys []types.ProjectKey) error {
+	var project types.Project
+	err, found := k.projectsFS.FindEntry(ctx, projectID, uint64(ctx.BlockHeight()), &project)
+	if err != nil || !found {
+		return utils.LavaError(ctx, ctx.Logger(), "AddProjectKeys_project_not_found", map[string]string{"project": projectID}, "project id not found")
+	}
+
+	// check if the admin key is valid
+	if !project.IsKeyType(adminKey, types.ProjectKey_ADMIN) && project.Subscription != adminKey {
+		return utils.LavaError(ctx, ctx.Logger(), "AddProjectKeys_not_admin", map[string]string{"project": projectID}, "the requesting key is not admin key")
+	}
+
+	// check that those keys are unique for developers
+	for _, projectKey := range projectKeys {
+		err = k.RegisterDeveloperKey(ctx, projectKey.Key, project.Index, uint64(ctx.BlockHeight()))
+		if err != nil {
+			return err
+		}
+
+		project.AppendKey(projectKey)
+	}
+
+	err = k.projectsFS.AppendEntry(ctx, projectID, uint64(ctx.BlockHeight()), &project)
```

Comment:
If there are 2 keys and one gets in but the other triggers an error, then this AppendEntry() will not be called - which is wrong.

Reply:
moot :)